### ISSUE TITLE
Have cidr_match() accept multiple subnets

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4494]
At the time of the filing of this issue, Zed is at commit 26783dc.

I noticed that the [`cidr_match()` function](https://zed.brimdata.io/docs/next/language/functions/cidr_match) already has a nice convenience in how it accepts complex types for its `val` argument in which case it will determine if any of the values it contains are within the range of the `mask` (subnet) argument. I then realized it would be a nice convenience if `mask` also accepted complex types such that `cidr_match()` could return true if one or more of the `val` values were within the range of _any_ of the multiple subnets within `mask`. This would be a clean solution to #1504.

In other words, whereas today we see this:

```
$ zq -version
Version: v1.7.0-18-g26783dcd

$ echo '10.1.2.129 11.1.2.129 192.168.0.1' | zq -z 'yield cidr_match(|[10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16]|, this)' -
error("cidr_match: not a net: |[10.0.0.0/8,172.16.0.0/12,192.168.0.0/16]|")
error("cidr_match: not a net: |[10.0.0.0/8,172.16.0.0/12,192.168.0.0/16]|")
error("cidr_match: not a net: |[10.0.0.0/8,172.16.0.0/12,192.168.0.0/16]|")
```

It could work like:

```
$ echo '10.1.2.129 11.1.2.129 192.168.0.1' | zq -z 'yield cidr_match(|[10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16]|, this)' -
true
false
true
```
